packages/core/src/graphql/index.ts

```diff
@@ -266,36 +266,46 @@ export function buildGraphQLSchema({
             };
           } else if (is(relation, Many)) {
             // Search the relations of the referenced table for the corresponding `one` relation.
-            // If "relationName" is not provided, use the first `one` relation that references this table.
+            // If `relation.relationName` is not provided, use the first `one` relation that references this table.
 
             let oneRelation: One | undefined;
 
-            for (const _relation of Object.values(referencedTable.relations)) {
-              if (
-                is(_relation, One) &&
-                relation.relationName === _relation.relationName
-              ) {
-                oneRelation = _relation;
+            // Note: can find the wrong relation if `relationName` is undefined.
+            // The first relation will be found.
+
+            if (relation.relationName !== undefined) {
```

Comment:
@typedarray The root of this bug is using `relationName` to try and find the corresponding "one" relation without first checking if `relationName` is `undefined`.

If `relationName` is not used in the relations (commonly is not) then the first comparison will return true because both `relation.relationName` and `referencedRelation.relationName` will be `undefined`.